### ISSUE TITLE
JSON object names need to be strings

### ISSUE THREAD

**Issue by rbaard1** [masaccio/compact-json#7]
First off: thanks for the very useful tool! I am really happy to have found it, it fills a gap in the standard library that I think [many people](https://stackoverflow.com/questions/21866774/pretty-print-json-dumps) [have experienced](https://stackoverflow.com/questions/16264515/json-dumps-custom-formatting).

The issue I encountered: in a JSON object, names can only be strings (see [2.2 in the RFC](https://www.ietf.org/rfc/rfc4627.txt)). So if you do:
```python
from compact_json import Formatter
Formatter().serialize({100: 'foobar'})
```
you should get either `{"100": "foo bar"}` (this is what the standard `json` library does), or perhaps raise an exception (this is what the `cjson` library [apparently does](https://stackoverflow.com/questions/5526995/why-does-python-json-module-represent-floats-in-dictionary-keys-as-strings#comment6670099_5527017)). But with `compact-json`, this gives  `{100: "foo bar"}`, which is invalid JSON. Trying to decode it with
```python
import json
json.loads('{100: "foo bar"}')
```
raises a `json.decoder.JSONDecodeError`: "Expecting property name enclosed in double quotes: line 1 column 2 (char 1)".

I think raising an exception is the clearest option here, since silently stringifying the key could lead to unexpected behavior. An optional `stringify_names` argument could be nice though.

Side note:, I think the `json` library is actually not handling this case correctly: `json.dumps({100: "foo", "100": "bar"}` gives `{"100": "foo", "100": "bar"}` which is invalid JSON since names should be unique.

**Comment by masaccio** [comment]
I've added a silent conversion of non-string keys to strings which mirrors the behaviour of the standard package. I was on the fence about whether to issue a runtime warning, but I decided that I'd replicate the silent behaviour of the standard package to ensure that the JSON is valid.

I agree @ZeroRin that the behaviour regarding unique keys is broken in the standard package and also in `compact-json`.

```
>>> json.dumps({int(100): "foo", float(100): "bar"})
'{"100": "bar"}'
>>> json.dumps({int(100): "foo", str(100): "bar"})
'{"100": "foo", "100": "bar"}'
```

If the intent is well-formed JSON (surely it is!) then I do think that unique keys plus a runtime warning is right here.
